nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/headers/ContentSecurityPolicyFilter.java

```diff
@@ -28,7 +28,7 @@
 import javax.servlet.FilterConfig;
 
 /**
- * A filter to apply the Content Security Policy (which supersedes the X-Frame-Options header).
+ * A filter to apply the Content Security Policy header.
  *
  */
 public class ContentSecurityPolicyFilter implements Filter {
```

Comment:
What about the `default-src 'self'` header value? See [Essential HTTP Security Headers](https://pentest-tools.com/blog/essential-http-security-headers/) - Section 2 for more details.

Reply:
More details via MDN - [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).

Reply:
We can apply the `default-src 'self'` directive but it also requires adding `'unsafe-inline'`, otherwise our many inline scripts will be blocked by default. They're already allowed anyway, so I guess there's no problem with this.